api/python/tests/integration/test_packages.py

```diff
@@ -666,6 +666,61 @@ def test_s3_set_dir(self):
 
             list_object_versions_mock.assert_called_with('bucket', 'foo/')
 
+    def test_set_dir_update_policy(self):
+        """Verify building a package with update policy. """
+
+        # Test for local src dir
+        foodir = pathlib.Path("foo_dir")
+        bazdir = pathlib.Path("baz_dir")
+        foodir.mkdir()
+        bazdir.mkdir()
+        with open(foodir / 'bar', 'w') as fd:
+            fd.write(fd.name)
+        with open(bazdir / 'bar', 'w') as fd:
+            fd.write(''.join([str(i) for i in range(100)]))
+
+        pkg = Package()
+        pkg = pkg.set_dir("/", ".", meta={'name': 'test_meta'})
```

Comment:
If we're going to set the meta, we should read it out in the context of the test. e.g. check that metadata are preserved after a `set_dir`.